src/app/common/resolvers/search.resolver.ts

```diff
@@ -9,10 +9,28 @@ import { inject } from "@angular/core";
 import { ActivatedRouteSnapshot, ResolveFn } from "@angular/router";
 import ProjectLinks from "src/app/project-links";
 import { SearchService } from "src/app/search/search.service";
+import { Observable, of, switchMap } from "rxjs";
+import { DatasetSearchResult } from "src/app/interface/search.interface";
+import { LoggedUserService } from "src/app/auth/logged-user.service";
+import { AppConfigService } from "src/app/app-config.service";
 
-export const searchResolver: ResolveFn<void> = (route: ActivatedRouteSnapshot) => {
+export const searchResolver: ResolveFn<Observable<DatasetSearchResult>> = (route: ActivatedRouteSnapshot) => {
     const searchService = inject(SearchService);
+    const loggedUserService = inject(LoggedUserService);
+    const appConfigService = inject(AppConfigService);
+    const isAuthenticated = loggedUserService.isAuthenticated;
+    const isSemanticSearchAvailable = appConfigService.semanticSearchScore;
     const page = route.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_PAGE) ?? 1;
     const query = route.queryParamMap.get(ProjectLinks.URL_QUERY_PARAM_QUERY) ?? "";
-    return searchService.searchDatasets(query, Number(page) - 1);
+
+    return searchService.searchDatasets(query, Number(page) - 1).pipe(
+        switchMap((result: DatasetSearchResult) => {
+            return result.datasets.length ||
+                !query ||
+                !isAuthenticated ||
+                typeof isSemanticSearchAvailable === "undefined"
```

Comment:
Exactly what I'm talking about. Boolean should be "true" or "false". As soon as you add "undefined", await for bugs and weird expressions that could be avoided.

Reply:
Fixed